modules/tasks/tasks/src/main/java/org/motechproject/tasks/util/SchedulerTaskTriggerUtil.java

```diff
@@ -0,0 +1,203 @@
+                }
+                break;
+
+            case REPEATING_JOB:
+                LOGGER.info("Scheduling repeating job for task: {}", task);
+                startDate = DATE_FORMAT.parseDateTime(trigger.getStartDate());
+                endDate = DATE_FORMAT.parseDateTime(trigger.getEndDate());
+                interval = trigger.getInterval();
+                ignorePastFiresAtStart = trigger.isIgnoreFiresignorePastFiresAtStart();
+                schedulerService.safeScheduleRepeatingJob(
+                        new RepeatingSchedulableJob(motechEvent, interval, startDate.toDate(), endDate.toDate(),
+                                ignorePastFiresAtStart));
+                break;
+
+            case CRON_JOB:
+                LOGGER.info("Scheduling cron job for task: {}", task);
+                startDate = DATE_FORMAT.parseDateTime(trigger.getStartDate());
+                endDate = DATE_FORMAT.parseDateTime(trigger.getEndDate());
+                cronExpression = trigger.getCronExpression();
+                ignorePastFiresAtStart = trigger.isIgnoreFiresignorePastFiresAtStart();
+                schedulerService.safeScheduleJob(
+                        new CronSchedulableJob(motechEvent, cronExpression, startDate.toDate(), endDate.toDate(),
+                                ignorePastFiresAtStart));
+                break;
+
+            case DAY_OF_WEEK_JOB:
+                LOGGER.info("Scheduling day of week job for task: {}", task);
+                startDate = DATE_FORMAT.parseDateTime(trigger.getStartDate());
+                endDate = DATE_FORMAT.parseDateTime(trigger.getEndDate());
+                days = trigger.getDays();
+                time = trigger.getTime();
+                ignorePastFiresAtStart = trigger.isIgnoreFiresignorePastFiresAtStart();
+                schedulerService.scheduleDayOfWeekJob(
+                        new DayOfWeekSchedulableJob(motechEvent, startDate.toLocalDate(), endDate.toLocalDate(),
+                                days, time, ignorePastFiresAtStart));
+                break;
+
+            case REPEATING_JOB_WITH_PERIOD_INTERVAL:
+                LOGGER.info("Scheduling repeating job with period interval for task: {}", task);
+                startDate = DATE_FORMAT.parseDateTime(trigger.getStartDate());
+                endDate = DATE_FORMAT.parseDateTime(trigger.getEndDate());
+                repeatPeriod = trigger.getRepeatPeriod();
+                ignorePastFiresAtStart = trigger.isIgnoreFiresignorePastFiresAtStart();
+                schedulerService.safeScheduleRepeatingPeriodJob(
+                        new RepeatingPeriodSchedulableJob(motechEvent, startDate.toDate(), endDate.toDate(),
+                                repeatPeriod, ignorePastFiresAtStart));
+            default:
+                break;
+        }
+    }
+
+    public void setSchedulerTaskTriggerType(Task task) {
+        String[] name = task.getTrigger().getSubject().split("\\.");
```

Comment:
Should create a method in this util that extracts the task name from the subject